File: Derrick_Kuhn_Code/Project_1/src/main/Java/com/revature/model/Request.java
package com.revature.model;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.Objects;

public class Request {
    private long r_id;
    private long e_id;
    private String subject;
    private String description;
    private double amount;
    private long approver_id;
    private Status status;

    public Request() {
    }

    public Request(long r_id, long approver_id, Status status){
        this.r_id = r_id;
        this.approver_id = approver_id;
        this.status = status;
    }

    public Request(long e_id, String subject, String description, double amount) {
        this.e_id = e_id;
        this.subject = subject;
        this.description = description;
        this.amount = amount;
        this.status = Status.PENDING;
    }

    public Request(long id, long e_id, String subject, String description, double amount, long approver_id, Status status) {
        this.r_id = id;
        this.e_id = e_id;
        this.subject = subject;
        this.description = description;
        this.amount = amount;
        this.approver_id = approver_id;
        this.status = status;
    }

    public long getR_id() {
        return this.r_id;
    }

    public void setR_id(long id) {
        this.r_id = id;
    }

    public long getE_id() {
        return e_id;
    }

    public void setE_id(long e_id) {
        this.e_id = e_id;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public long getApprover_id() {
        return approver_id;
    }

    public void setApprover_id(long approver_id) {
        this.approver_id = approver_id;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Request request = (Request) o;
        return r_id == request.r_id &&
                e_id == request.e_id &&
                Double.compare(request.amount, amount) == 0 &&
                approver_id == request.approver_id &&
                Objects.equals(subject, request.subject) &&
                Objects.equals(description, request.description) &&
                status == request.status;
    }

    @Override
    public int hashCode() {
        return Objects.hash(r_id, e_id, subject, description, amount, approver_id, status);
    }

    @Override
    public String toString() {
        return "Request{" +
                "r_id=" + r_id +
                ", e_id=" + e_id +
                ", subject='" + subject + '\'' +
                ", description='" + description + '\'' +
                ", amount=" + amount +
                ", approver_id=" + approver_id +
                ", status=" + status +
                '}';
    }
}

File: Derrick_Kuhn_Code/Project_1/src/main/webapp/javascript/Request.js
import Employee from "./Employee";

class Request {

    constructor(name) {
        // invokes the setter
        this._r_id = 0;
        this._e_id = 0;
        this._subject = 'subject';
        this._description = 'description';
        this._amount = 12.34;
        this._approver_id = 0;
        this._status = 'PENDING';
        this._name = name;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }

    get r_id() {
        return this._r_id;
    }

    set r_id(value) {
        this._r_id = value;
    }

    get e_id() {
        return this._e_id;
    }

    set e_id(value) {
        this._e_id = value;
    }

    get subject() {
        return this._subject;
    }

    set subject(value) {
        this._subject = value;
    }

    get description() {
        return this._description;
    }

    set description(value) {
        this._description = value;
    }

    get amount() {
        return this._amount;
    }

    set amount(value) {
        this._amount = value;
    }

    get approver_id() {
        return this._approver_id;
    }

    set approver_id(value) {
        this._approver_id = value;
    }

    get status() {
        return this._status;
    }

    set status(value) {
        this._status = value;
    }
}

export default Request;